lib/templates/src/components/Chat.vue

```diff
@@ -0,0 +1,42 @@
+<template>
+  <div>
+    <div id="chat-box">
+      <p v-for="message in messages" :key="message" v-text="message"></p>
+    </div>
+    <div class="magpie-view-answer-container">
+      <textarea
+        ref="text"
+        cols="50"
+        placeholder="Type your message to the other participant here."
+      ></textarea>
+      <button @click.stop="send()">
+        Send
+      </button>
+    </div>
+  </div>
+</template>
+
+<script>
+const EVENT_CHAT_MESSAGE = 'chat_message';
+
+export default {
+  name: 'Chat',
+  data() {
+    return {
+      messages: []
+    };
+  },
+  socket: {
+    [EVENT_CHAT_MESSAGE](payload) {
```

Comment:
Could you explain a bit more how this `socket: { [EVENT_CHAT_MESSAGE](payload) { } }` syntax works (guess this is also what vue-socket is doing)? It's super convenient though I feel that some users might feel lost without some clarification, especially if they're not that familiar with JS. Maybe you could add a comment or write it up in a documentation somewhere.

Reply:
Yes, I will definitely document that!

The socket object as part of the component definition sets up subscriptions for specific socket events when the component is mounted, the callbacks of which will be called when the event is received via the socket.